### PR TITLE
Test active project when calling `test` without positional arguments

### PR DESCRIPTION
**Pull request by bauglir**
This addresses #14.

In the process, I have slightly altered the related `testset`s by introducing some helper functions and additional `testset`s for clarity. I tend to do this on my own projects so test run output is easier to parse, but please let me know if you'd prefer a different approach.

I also realize there may be some overlap on the test side with #10 and #13, let me know if there's anything I can do to coordinate that better.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -57,18 +57,54 @@ end
...
+            @test_throws TestReports.PkgTestError TestReports.test(TEST_PKG.name)
+        end
+
+        @testset "implicitly without positional arguments" begin
+            # The test run should not fail when a project is implied through
+            # one being active and fail with a descriptive error otherwise
+            with_active_project(TEST_PKG.name) do
+                TestReports.test()
+            end
+            @test_throws TestReports.PkgTestError TestReports.test()
```
Comment (bauglir) [inline test/runtests.jl:107]: It looks like [#15.2](https://travis-ci.com/github/JuliaTesting/TestReports.jl/jobs/363647241) fails as this test is assumed to be run in the context where no project is active. However, on Travis deactivating the `Example` package/project reactivates the `TestReports` package/project which causes the job to go into a loop and fail.

Comment (bauglir) [reply test/runtests.jl:107]: This is caused by the presence of the [`JULIA_PROJECT` environment variable](https://travis-ci.com/github/JuliaTesting/TestReports.jl/jobs/363647241#L185), which makes a plain `Pkg.activate()` revert back to that specific project. I've been able to reproduce the behavior locally.

The easiest 'solution' is to drop the test for 'no active project' case, but I don't think that is the right thing to do here. I'll investigate further.

Comment (bauglir) [reply test/runtests.jl:117]: Resolved in 04fc4d276b740f5ed3fcbe7118cc1fa6c0e80ad6.

**Comment by bauglir** [comment]
This consists of two commits now to address an oversight in the first one. Let me know if you'd like me to squash the two!

**Comment by mmiller-max** [comment]
Thanks for this!

> I have slightly altered the related testsets by introducing some helper functions and additional testsets for clarity

Love the `test_successful_testrun` function, was looking for a way of testing the output.

> I also realize there may be some overlap on the test side with #10 and #13, let me know if there's anything I can do to coordinate that better.

Yep there's a small bit of overlap. I'm aiming to use `temp_pkg_dir` from `utils.jl` in `Pkg.jl` tests, as it isolates all the tests packages nicely and returns to the original state without needing a `Pkg.activate()` each time. I think this will supersede some of the use cases for `with_active_project`, for example

```Julia
temp_pkg_dir() do tmp
    @test_throws TestReports.PkgTestError TestReports.test()
end
```

should work as `JULIA_PROJECT` is set to `nothing` within  `temp_pkg_dir`. I'd like to keep the approach here as aligned as possible with `Pkg`.

However I'm still working on these changes so I think it makes sense to get this one in and go from there, if what I've said above makes sense? If so please squash and we can merge.

**Comment by bauglir** [comment]
Thanks for the review. I've just squashed and pushed the fix as a single commit.

Using the `temp_pkg_dir` function from `Pkg.jl` indeed sounds like the better approach, but I'll let you figure that out first :) I wonder how feasible it would be to straight-up reuse these functions. Which I guess is a similar query as I posed in the corresponding issue. It would be a shame to have to copy all those useful utility functions over. We can figure that out in a separate context though.

Having this merged would help me out so I can update my CI/CD pipelines so they're able to output JUnit reports again, so that merge would be highly appreciated.

**Comment by mmiller-max** [comment]
Sorry one last change, please can you bump the version to v0.2.1?

**Comment by bauglir** [comment]
I've added the bumped version to the squashed commit.

**Pull request merged**